code/iaas/auth-logic/src/main/java/io/cattle/platform/iaas/api/auth/github/ProjectResourceManager.java

```diff
@@ -54,148 +57,84 @@
 
     @Override
     public Class<?>[] getTypeClasses() {
-        return new Class[] {};
+        return new Class<?>[0];
     }
 
     @Override
     public Object listInternal(SchemaFactory schemaFactory, String type, Map<Object, Object> criteria, ListOptions options) {
-        ApiRequest apiRequest = ApiContext.getContext().getApiRequest();
-        String token = getTokenFromRequest(apiRequest);
-        GithubUtils.AccesibleIds ids = githubUtils.validateAndFetchAccesibleIdsFromToken(token);
-        String id = (String) criteria.get("id");
-        if (StringUtils.isNotEmpty(id)) {
+        Policy policy = (Policy) ApiContext.getContext().getPolicy();
+        Object obj = criteria.get("id");
+        String id = "";
+        if (obj != null) {
+            if (obj instanceof String) {
+                id = (String) obj;
+            }else {
+                throw new ClientVisibleException(ResponseCodes.NOT_FOUND);
+            }
             Account account = getObjectManager().loadResource(Account.class, id);
```

Comment:
To avoid confusion, I wonder should this variable be called 'project' instead of 'account'